src/iosource/pcap/Source.cc

```diff
@@ -176,7 +178,32 @@ void PcapSource::OpenOffline()
 	{
 	char errbuf[PCAP_ERRBUF_SIZE];
 
-	pd = pcap_open_offline(props.path.c_str(), errbuf);
+	FILE* f = nullptr;
+	if ( props.path == "-" )
+		{
+		f = stdin;
+		}
+	else
+		{
+		if ( ! (f = fopen(props.path.c_str(), "rb")) )
+			{
+			Error(util::fmt("unable to open %s: %s", props.path.c_str(), strerror(errno)));
+			return;
+			}
+
+		if ( BifConst::Pcap::bufsize_offline_bytes != 0 )
+			{
+			iobuf.resize(BifConst::Pcap::bufsize_offline_bytes);
+			if ( util::detail::setvbuf(f, iobuf.data(), _IOFBF, iobuf.size()) != 0 )
+				{
+				Error(util::fmt("unable to setvbuf %s: %s", props.path.c_str(), strerror(errno)));
+				fclose(f);
+				return;
+				}
+			}
+		}
+
+	pd = pcap_fopen_offline(f, errbuf);
```

Comment:
This currently leaks `f` since we do not seem to call `::fclose` on it for the success path.

Reply:
`pcap_fopen_offline` takes ownership of `f`.

Reply:
That seems to be undocumented behavior, https://linux.die.net/man/3/pcap_open_offline.

Reply:
Actually - in the error case (when there's no actual pcap content in the file) we're leaking `f` I suspect. Let me see if this triggers on a sanitizer build.